### DISCUSSION TITLE
Handling password fallback

### DISCUSSION THREAD

**Discussion by johnbaker0583** [jadolg/rocketchat_API#131] [Ideas]
Would it be possible to include this for settings updates?

https://developer.rocket.chat/api/rest-api/endpoints/team-collaboration/2fa 

If an api request returns TOTP Required with a method password, then the API user's password is required to authenticate the request:

// Error example
{
  "success":false,
  "error":"TOTP Required [totp-required]",
  "errorType":"totp-required",
  "details":{
    "method":"password",
    "codeGenerated":false,
    "availableMethods":[]
  }
}

**Comment by jadolg** [comment]
I have not messed with TOPT so far because I have not found a good way of testing automatically that it works.
